### PR TITLE
feature/XTG-235 add submit score endpoint

### PR DESCRIPTION
**Pull request by fredericobraga85**
### Ticket

ticket: https://x-team-internal.atlassian.net/browse/XTG-235

### Description

adding submit endpoint for webhook
fixed some bugs
cleaned some code
adjusted tests

### CODE REVIEW

File: src/api-utils/responseSchemas/gamedev.ts [hunk 1]
```diff
@@ -2,6 +2,38 @@ import Joi from 'joi';
...
+      Joi.object({
+        attribute: Joi.string().required(),
+        value: Joi.string().required(),
+      })
+    )
+    .optional(),
+  createdAt: Joi.date(),
+  updatedAt: Joi.date(),
+}).required();
+
```
Comment (fredericobraga85) [inline src/api-utils/responseSchemas/gamedev.ts:36]: I have the feeling that this file will become pretty in the future.. we should probably start thinking in spliting it

File: src/models/LeaderboardResults.ts [hunk 2]
```diff
@@ -69,18 +73,12 @@ export class LeaderboardResults
...
-
   @AllowNull(false)
   @Column(DataType.INTEGER)
   score!: number;
 
-  @AllowNull(true)
-  @Column(DataType.JSON)
-  meta!: JSON;
+  @HasMany(() => LeaderboardResultsMeta, '_leaderboardResultsId')
+  _leaderboardResultsMeta?: LeaderboardResultsMeta[];
```
Comment (fredericobraga85) [inline src/models/LeaderboardResults.ts:81]: adding the new leaderboardResultsMeta association

File: src/models/LeaderboardResults.ts [hunk 3]
```diff
@@ -91,18 +89,52 @@ export class LeaderboardResults
...
+            ...meta,
+            _leaderboardResultsId: r[0].id,
+          },
+          { transaction }
+        );
+      }
+    }
+
+    return r;
+  });
```
Comment (fredericobraga85) [inline src/models/LeaderboardResults.ts:125]: I have some mixed feelings on how I implemented this..
please share your thoughts if you have a more elegant approach

File: src/modules/gameDevs/gameDevWebhookHandler.ts [hunk 4]
```diff
@@ -10,3 +18,33 @@ export const getAchievementsThruWebhookHandler: Lifecycle.Method = async (reques
   // }
   return h.response({ achievements }).code(200);
 };
+
+export const postLeaderboardResultHandler: Lifecycle.Method = async (request, h) => {
+  const { gameType } = request.pre.webhookValidation;
+  const payload = request.pre.webhookPayload as LeaderboardResultsCreationAttributes;
```
Comment (fredericobraga85) [inline src/modules/gameDevs/gameDevWebhookHandler.ts:24]: payload is coming from the webhook payload parser as we mentioned on the call.
the schema validation occurs inside that middleware

File: src/modules/gameDevs/utils.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  const body = request.payload.toString('utf-8');
+  const payload = JSON.parse(body);
+
+  //TBD: add validation for new requests
+  const validationRslt = postLeaderboardResultScoreSchema.validate(payload);
+
+  checkForWebhookErrors(validationRslt, payload);
+
+  return payload;
+};
```
Comment (fredericobraga85) [inline src/modules/gameDevs/utils.ts:21]: to note: so far the parsewebhookpayload middleware only validates the postLeaderboardResultScoreSchema

Comment (ccmoralesj) [reply src/modules/gameDevs/utils.ts:21]: Every new request will need its own validation, just like we have with Slack.

File: src/server.ts [hunk 6]
```diff
@@ -43,7 +43,16 @@ const getServer = () =>
...
+        headers: [
+          'Accept',
+          'Authorization',
+          'Content-Type',
+          'If-None-Match',
+          'xtu-session-token',
+          'xtu-client-secret',
+          'xtu-request-timestamp',
+          'xtu-signature',
+        ],
```
[the remark this answers is not in the document]
Comment (ccmoralesj) [reply src/server.ts:55]: Thanks a lot!

**Review by ccmoralesj — CHANGES_REQUESTED** [review]

**Review by ccmoralesj — APPROVED** [review]

**Pull request merged**